windows/src/engine/kmtip/tmgrsink.cpp

```diff
@@ -240,17 +257,23 @@ BOOL CKMTipTextService::_InitThreadMgrSink()
 
 void CKMTipTextService::_UninitThreadMgrSink()
 {
-    ITfSource *pSource;
-
-    if (_pThreadMgr->QueryInterface(IID_ITfSource, (void **)&pSource) == S_OK)
-    {
-      if (_dwThreadMgrEventSinkCookie != TF_INVALID_COOKIE)
-        pSource->UnadviseSink(_dwThreadMgrEventSinkCookie);
-      if (_dwActiveLanguageProfileNotifySinkCookie != TF_INVALID_COOKIE)   // I3581
-        pSource->UnadviseSink(_dwActiveLanguageProfileNotifySinkCookie);
-      pSource->Release();
+  LogEnter();
+
+  ITfSource *pSource;
+
+  HRESULT res = _pThreadMgr->QueryInterface(IID_ITfSource, (void **)&pSource);
+  if(res == S_OK) {
+    if (_dwThreadMgrEventSinkCookie != TF_INVALID_COOKIE) {
+      res = pSource->UnadviseSink(_dwThreadMgrEventSinkCookie);
+      if (res != S_OK) DebugLastError0(L"UnadviseSink(ThreadMgrEventSink)", res);
+    }
+    if (_dwActiveLanguageProfileNotifySinkCookie != TF_INVALID_COOKIE) {  // I3581
+      pSource->UnadviseSink(_dwActiveLanguageProfileNotifySinkCookie);
```

Comment:
Should this be
```
res = pSource->UnadviseSink(_dwActiveLanguageProfileNotifySinkCookie);
``` 
(like line 267)

Reply:
Fixed (might want to review with "ignore whitespace diffs" now!)